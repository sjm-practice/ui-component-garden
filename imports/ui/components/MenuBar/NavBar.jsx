import React from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";
import { withStyles } from "@material-ui/core/styles";

const pathAndHash = (match, location, toHash) => {
  if (!match) {
    return false;
  }

  return match.url === location.pathname && toHash === location.hash;
};

const styles = theme => ({
  selected: {
    color: theme.palette.primary.main,
    "&:hover": {
      color: theme.palette.primary.main,
    },
  },
});

// TODO: to set an anchor link active, could check location.hash
// if equal, set Contact to active
// Note, may need to use Link instead of NavLink to do so
const NavBar = ({ classes: { selected } }) => (
  <nav className="menubar-container menubar-middle">
    <NavLink
      exact
      to="/"
      activeClassName={selected}
      isActive={(m, l) => pathAndHash(m, l, "")}
    >
      HOME
    </NavLink>
    <NavLink
      exact
      to={{ pathname: "/", hash: "#contact" }}
      activeClassName={selected}
      isActive={(m, l) => pathAndHash(m, l, "#contact")}
    >
      CONTACT
    </NavLink>
    <NavLink
      exact
      to="/privacy-policy"
      activeClassName={selected}
      isActive={(m, l) => pathAndHash(m, l, "")}
    >
      PRIVACY
    </NavLink>
  </nav>
);

NavBar.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default withStyles(styles)(NavBar);
