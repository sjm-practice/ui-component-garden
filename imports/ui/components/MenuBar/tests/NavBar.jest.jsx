/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import NavBar from "../NavBar";

describe("<NavBar />", () => {
  it("should match render snapshot", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <NavBar />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("Selected Links", () => {
    it("should show HOME link selected", () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={["/"]}>
          <NavBar />
        </MemoryRouter>,
      );

      const home = wrapper.find("a[href='/']");
      expect(home.props().className).toMatch(/selected/);

      const contact = wrapper.find("a[href='/#contact']");
      expect(contact.props().className).toBeUndefined();

      const privacy = wrapper.find("a[href='/privacy-policy']");
      expect(privacy.props().className).toBeUndefined();
    });

    it("should show CONTACT link selected", () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={["/#contact"]}>
          <NavBar />
        </MemoryRouter>,
      );

      const home = wrapper.find("a[href='/']");
      expect(home.props().className).toBeUndefined();

      const contact = wrapper.find("a[href='/#contact']");
      expect(contact.props().className).toMatch(/selected/);

      const privacy = wrapper.find("a[href='/privacy-policy']");
      expect(privacy.props().className).toBeUndefined();
    });

    it("should show PRIVACY link selected", () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={["/privacy-policy"]}>
          <NavBar />
        </MemoryRouter>,
      );

      const home = wrapper.find("a[href='/']");
      expect(home.props().className).toBeUndefined();

      const contact = wrapper.find("a[href='/#contact']");
      expect(contact.props().className).toBeUndefined();

      const privacy = wrapper.find("a[href='/privacy-policy']");
      expect(privacy.props().className).toMatch(/selected/);
    });
  });
});
