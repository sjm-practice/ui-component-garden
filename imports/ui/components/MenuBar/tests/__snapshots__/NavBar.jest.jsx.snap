// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match render snapshot 1`] = `
<nav
  className="menubar-container menubar-middle"
>
  <a
    aria-current="page"
    className="NavBar-selected-1"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    HOME
  </a>
  <a
    aria-current={null}
    href="/#contact"
    onClick={[Function]}
  >
    CONTACT
  </a>
  <a
    aria-current={null}
    href="/privacy-policy"
    onClick={[Function]}
  >
    PRIVACY
  </a>
</nav>
`;
